request_test.go

```diff
@@ -1237,6 +1238,76 @@ func TestRequest_BodyMultipart(t *testing.T) {
 		eof, _ := reader.NextPart()
 		assert.Nil(t, eof)
 	})
+
+	t.Run("multipart writer error", func(t *testing.T) {
+		cases := []struct {
+			name     string
+			reqFn    func() *Request
+			assertFn func(t *testing.T, req *Request)
+		}{
+			{
+				name: "with form",
+				reqFn: func() *Request {
+					req := NewRequestC(config, "POST", "url")
+					req.multipartFn = func(w io.Writer) *multipart.Writer {
+						return multipart.NewWriter(&failureWriter{})
+					}
+					req.WithMultipart()
```

Comment:
Seems this part:

```go
					req := NewRequestC(config, "POST", "url")
					req.multipartFn = func(w io.Writer) *multipart.Writer {
						return multipart.NewWriter(&failureWriter{})
					}
					req.WithMultipart()
```

is same for all subtests, so it can be extracted to the loop. And reqFn can accept req as argument instead of returning req.

Reply:
updated https://github.com/gavv/httpexpect/pull/344/commits/04042a6e44d71805c60bd675cf13ff067f5db142